RELEASE_NOTES.md

```diff
@@ -0,0 +1,5 @@
+#### 1.0.1 May 27 2016 ####
+Removed non-generic IPublisher and ISubscriber interfaces.
```

Comment:
1. also done (updated the PR).
Working on 3.

OK to merge this so we can add the TCK with a separate PR? If not, how would you prefer to proceed?